### PR TITLE
Fix default values on variables interpolation

### PR DESCRIPTION
**Pull request by ulyssessouza**
Handles extra closing braces by distinguishing it's chooser from the rest of the braces

Resolves https://github.com/docker/compose/issues/9024
Resolves https://github.com/docker/compose/issues/8799

### CODE REVIEW

**Comment by i-ky** [comment]
@ulyssessouza, does this fix address `:?` and `?` syntax as well? I am asking because tests only feature `:-` and `-` syntax.

**Comment by ulyssessouza** [comment]
@i-ky Just added some tests for required. Thanks!

File: template/template.go [hunk 1]
```diff
@@ -100,7 +103,11 @@ func SubstituteWith(template string, mapping Mapping, pattern *regexp.Regexp, su
 				if !applied {
 					continue
 				}
-				return value
+				interpolatedNested, err := SubstituteWith(rest, mapping, pattern, subsFuncs...)
+				if err != nil {
+					return ""
+				}
+				return value + interpolatedNested
```
Comment (glours) [inline template/template.go:110]: nit: 
```suggestion
				if interpolatedNested, err := SubstituteWith(rest, mapping, pattern, subsFuncs...); err == nil {
					return value + interpolatedNested
				}
				return ""
```

**Review by glours — APPROVED** [review]
LGTM
Only a little nit, but we can merge without fixing it

**Pull request merged**